### PR TITLE
Handle XDC replication task retry error / not found error correctly

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Handle XDC replication task retry error / not found error correctly
  * When encountered not found, delete workflow abort replication task
  * When encountered another retry error, retry one more time

**Why?**
Properly handle reset workflow & deleted workflow case

**How did you test it?**
UT

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: service/history/replication/executable_task.go [hunk 1]
```diff
@@ -316,14 +332,37 @@ func (e *ExecutableTaskImpl) Resend(
...
+		// 1. replicating a workflow which is a reset workflow (call this workflow `new workflow`)
+		// 2. base workflow (call this workflow `old workflow`) of reset workflow is deleted on
+		//	src cluster and never replicated to target cluster
+		// 3. when any of events of the new workflow arrive at target cluster
+		//  a. using base workflow info to resend until branching point between old & new workflow
+		//  b. attempting to use old workflow history events to replay for mutable state then apply new workflow events
+		//  c. attempt failed due to old workflow does not exist
+		//  d. return error to resend new workflow before the branching point
+
+		// handle 2nd resend error, then 1st resend error
```
Comment (hehaifengcn) [inline service/history/replication/executable_task.go:354]: can you entail what this comment mean?

Comment (wxing1292) [reply service/history/replication/executable_task.go:354]: src cluster:
T = 0 base workflow -> new workflow due to reset
T = 1 base workflow deleted due to retention
target cluster (brand new):
T = 2 receiving replication task for new workflow (e.g. migration)
T = 3 attempting to resend history to the point where base workflow & new workflow diverge, e.g. event ID 100
T = 4 attempting to use base workflow to replay for event 1-99, then apply 100, however base workflow does not exists on target cluster
T = 5 resend 1-99 using new workflow run ID from src cluster

File: service/history/replication/executable_task.go [hunk 2]
```diff
@@ -316,14 +332,37 @@ func (e *ExecutableTaskImpl) Resend(
...
+		if _, err := e.Resend(ctx, remoteCluster, resendErr, remainingAttempt); err == nil {
```
Comment (yux0) [inline service/history/replication/executable_task.go:355]: Should we handle it in NDCHistoryResender?

**Review by meiliang86 — APPROVED** [review]

File: service/history/replication/executable_task.go (hunk 2 above)
Comment (wxing1292) [reply service/history/replication/executable_task.go:355]: there is another functionality which handles this case:
T = 0: received a replication task
T = 1: resend due to missing some events
T = 2: workflow deleted (due to retention) from source cluster
T = 3: above replication task & corresponding workflow should be deleted

we probably should (later) move these special case handling into a dedicated place

**Pull request merged**